src/base/global.h

```diff
@@ -26,5 +26,20 @@
  * exception statement from your version.
  */
 
+#pragma once
+
 const char C_TORRENT_FILE_EXTENSION[] = ".torrent";
 
+
+#if QT_VERSION < QT_VERSION_CHECK(5, 7, 0)
+template <typename T>
+constexpr typename std::add_const<T>::type &qAsConst(T &t) noexcept { return t; }
+
+// prevent rvalue arguments:
+template <typename T>
+void qAsConst(const T &&) = delete;
```

Comment:
AFAIK, it should be `qAsConst(T &&)` (without const).

Reply:
@glassez `qAsConst(const T &&)` also does the job.
And it guarantees the compatibility with Qt `qAsConst()`. Definition https://github.com/qt/qtbase/blob/5.11/src/corelib/global/qglobal.h#L960.

This way, when the minimum required Qt version reaches to 5.7 qBittorrent `qAsConst()` can safely be removed. No surprises.

p.s. I should have mentioned it.